indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+        self._styles["table_body"] = getSampleStyleSheet()["Normal"]
 
-        convenersStyle=getSampleStyleSheet()["Normal"]
+        convenersStyle = getSampleStyleSheet()["Normal"]
         convenersStyle.fontSize = modifiedFontSize(10.0, self._fontsize)
-        convenersStyle.leftIndent=10
-        self._styles["conveners"]=convenersStyle
+        convenersStyle.leftIndent = 10
+        self._styles["conveners"] = convenersStyle
 
-        subContStyle=getSampleStyleSheet()["Normal"]
-        subContStyle.fontSize=modifiedFontSize(10.0, self._fontsize)
-        subContStyle.leftIndent=15
-        self._styles["subContrib"]=subContStyle
+        subContStyle = getSampleStyleSheet()["Normal"]
+        subContStyle.fontSize = modifiedFontSize(10.0, self._fontsize)
+        subContStyle.leftIndent = 15
+        self._styles["subContrib"] = subContStyle
 
-    def _getSessionColor(self, ses):
-        HTMLcolor = ses.getSession().getColor()
-        color = html_color_to_rgb(HTMLcolor.strip())
-        return color
+    def _getSessionColor(self, block):
+        session_color = '#{}'.format(block.session.colors.background)
+        return html_color_to_rgb(session_color)
 
-    def _processContribution(self,contrib,l):
-        if not contrib.canAccess(self._aw):
+    def _processContribution(self, contrib, l):
+        if not contrib.can_access(self._aw):
             return
-        date = "%s"%escape(contrib.getAdjustedStartDate(self._tz).strftime("%H:%M"))
-        #date="<font size=\"" + str(modifiedFontSize(10, self._fontsize)) + "\">%s</font>"%date
-        #date=self._fontify(date,10)
-        #date.style=self._styles["table_body"]
-        date=Paragraph(date,self._styles["table_body"])
-        lt=[]
-        captionText="[%s] %s"%(escape(contrib.getId()),escape(contrib.getTitle()))
+
+        lt = []
+        date = Paragraph(format_time(contrib.start_dt, timezone=self._tz), self._styles["table_body"])
+        caption = '[{}] {}'.format(contrib.id, escape(contrib.title))
+
         if not self._ttPDFFormat.showContribId():
-            captionText="%s"%(escape(contrib.getTitle()))
-        if self._ttPDFFormat.showLengthContribs():
-            captionText="%s (%s)"%(captionText, escape((datetime(1900,1,1)+contrib.getDuration()).strftime("%Hh%M'")))
-        if self._ttPDFFormat.showContribAbstract():
-            captionText="<font name=\"Times-Bold\"><b>%s</b></font>"%captionText
-        captionText="<font size=\"" + str(modifiedFontSize(10, self._fontsize)) + "\">%s</font>"%captionText
-        lt.append([self._fontify(captionText,10)])
-        colorCell = ""
+            caption = escape(contrib.title)
+        elif self._ttPDFFormat.showLengthContribs():
+            caption = "{} ({})".format(caption, format_human_timedelta(contrib.timetable_entry.duration))
+        elif self._ttPDFFormat.showContribAbstract():
+            caption = '<font name="Times-Bold"><b>{}</b></font>'.format(caption)
```

Comment:
I think it's used for the PDF generation so it's pseudo-HTML. `<font name=..>` was never valid HTML (the attribute was called `face`)